crates/biome_js_analyze/src/analyzers/style/use_consistent_array_type.rs

```diff
@@ -0,0 +1,485 @@
+        let query = ctx.query();
+        let mut mutation = ctx.root().begin();
+
+        match query {
+            Query::TsReferenceType(ty) => {
+                mutation.replace_node(AnyTsType::TsReferenceType(ty.clone()), state.clone());
+                if let Some(kind) = get_array_kind_by_reference(ty) {
+                    return Some(JsRuleAction {
+                        category: ActionCategory::QuickFix,
+                        applicability: Applicability::MaybeIncorrect,
+                        message: get_action_message(kind),
+                        mutation,
+                    });
+                }
+            }
+            Query::TsTypeAnnotation(ty) => {
+                let ty = ty.ty().ok()?;
+                if let AnyTsType::TsTypeOperatorType(opt_ty) = &ty {
+                    mutation
+                        .replace_node(AnyTsType::TsTypeOperatorType(opt_ty.clone()), state.clone());
+                    if let Some(kind) = get_array_kind_by_any_type(&ty) {
+                        return Some(JsRuleAction {
+                            category: ActionCategory::QuickFix,
+                            applicability: Applicability::MaybeIncorrect,
+                            message: get_action_message(kind),
+                            mutation,
+                        });
+                    }
+                } else if let AnyTsType::TsArrayType(array_type) = &ty {
+                    mutation
+                        .replace_node(AnyTsType::TsArrayType(array_type.clone()), state.clone());
+                    if let Some(kind) = get_array_kind_by_any_type(&ty) {
+                        return Some(JsRuleAction {
+                            category: ActionCategory::QuickFix,
+                            applicability: Applicability::MaybeIncorrect,
+                            message: get_action_message(kind),
+                            mutation,
+                        });
+                    }
+                }
+            }
+        }
+
+        None
+    }
+}
+
+fn get_array_kind_by_reference(ty: &TsReferenceType) -> Option<TsArrayKind> {
+    let name = ty.name().ok()?;
+    name.as_js_reference_identifier().and_then(|ident| {
+        let name = ident.value_token().ok()?;
+        match name.text_trimmed() {
+            "Array" => Some(TsArrayKind::GenericArray),
+            "ReadonlyArray" => Some(TsArrayKind::ReadonlyGenericArray),
+            _ => None,
+        }
+    })
+}
+
+fn get_array_kind_by_any_type(ty: &AnyTsType) -> Option<TsArrayKind> {
+    if let AnyTsType::TsTypeOperatorType(ty) = ty {
+        let operator_token = ty.operator_token().ok()?;
+        let is_readonly = operator_token.text_trimmed() == "readonly";
+
+        if let AnyTsType::TsArrayType(_) = ty.ty().ok()? {
+            if is_readonly {
+                return Some(TsArrayKind::Readonly);
+            }
+        }
```

Comment:
This could be simplified to something like:

```suggestion
        (matches!(ty.ty(), Ok(AnyTsType::TsArrayType(_))) && is_readonly).then_some(TsArrayKind::Readonly)
```

`matches!()` can be used in place of `if let` when you don't need to bind a variable.